libs/platform/kernel/registry_helper_km.c

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) Microsoft Corporation
+// SPDX-License-Identifier: MIT
+
+#include "ebpf_registry_helper.h"
+
+void
+close_registry_key(ebpf_registry_key_t key)
+{
+    ZwClose(key);
```

Comment:
We might want to track and return the return value of this API.  A failure here indicates a serious system error that needs to be indicated to the caller (and hopefully gets logged).  If I remember right, this call typically flushes the registry updates (and thus ensures data persistence).  A failure here could imply lost data which might be critical.

Reply:
This issue applies also to `\libs\api_common\registry_helper.cpp`, as this new `libs/platform/kernel/registry_helper_km.c` just separates what formerly fully embedded in `\include\user\ebpf_registry_helper.h`.

Given the scope of this PR (i.e. wrapping direct calls to the registry APIs to store APIs), I filed issue #2123 to harden all registry APIs.

Reply:
I'd disagree again for the exact same reasons mentioned in my feedback re. the 'string length' issue I'd pointed out earlier.

Reply:
Same as [here](https://github.com/microsoft/ebpf-for-windows/pull/2114#discussion_r1113956256)